core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/jobs/RemoteJobTracker.java

```diff
@@ -203,11 +168,84 @@ private void update(String jobId, String script, Map<String, Object> params) {
 		});
 		notifyChanged(jobId);
 	}
+	
+	private void convertSuspendedJobStatuses() {
+		
+		this.index.write(writer -> {
+			final Expression filter = RemoteJobEntry.Expressions.states(Set.of(RemoteJobState.SCHEDULED, RemoteJobState.RUNNING, RemoteJobState.CANCEL_REQUESTED));
+			final BulkUpdate<RemoteJobEntry> update = new BulkUpdate<>(
+				RemoteJobEntry.class, 
+				filter, 
+				RemoteJobEntry.WITH_DONE,
+				ImmutableMap.of("state", RemoteJobState.FAILED.name(), "finishDate", System.currentTimeMillis())
+			);
+			writer.bulkUpdate(update);
+			writer.commit();
+			return null;
+		});
+		
+	}
+
+	private int getNumberOfCompletedJobs() {
+		
+		return index.read(searcher -> {
+			return searcher.search(
+				Query.select(RemoteJobEntry.class)
+					.fields(RemoteJobEntry.Fields.STATE)
+					.where(Expressions.builder()
+							.filter(RemoteJobEntry.Expressions.done())
+							.build())
+					.limit(0)
+					.build()
+			);
+		})
+		.getTotal();
+		
+	}
+
+	@VisibleForTesting
+	public RemoteJobEntry get(String jobId) {
+		return index.read(searcher -> searcher.get(RemoteJobEntry.class, jobId));
+	}
+
+	@VisibleForTesting
+	public int getCompletedJobCounter() {
+		return completedJobCounter.get();
+	}
+
+	private void purge() {
+		
+		index.write(writer -> {
+			final Hits<String> hits = writer.searcher().search(Query.select(String.class)
+					.fields(RemoteJobEntry.Fields.ID)
+					.from(RemoteJobEntry.class)
+					.where(
+						Expressions.builder()
+							.must(RemoteJobEntry.Expressions.done())
+							.must(Expressions.builder()
+									.should(RemoteJobEntry.Expressions.deleted(true))
+									.should(RemoteJobEntry.Expressions.finishDate(0L, System.currentTimeMillis() - staleJobAge))
+									.build()
+							)
+							.build()
+					)
+					.limit(Integer.MAX_VALUE)
+					.build());
+			if (hits.getTotal() > 0) {
+				LOG.trace("Purging deleted / stale job entries {}", hits.getHits());
+				writer.remove(RemoteJobEntry.class, Set.copyOf(hits.getHits()));
+				writer.commit();
+			}
+			return null;
+		});
+		
+		completedJobCounter.set(getNumberOfCompletedJobs());
```

Comment:
The counter here should be set to `0` and used for counting the number of completed jobs, then triggering the purge when at least X (configurable threshold) jobs have completed since the last purge.